### PR TITLE
Fix color picker template

### LINKED ISSUE #21173: Adding a color picker throws an undefined index notice
#### Describe the bug

I made a module that has a color picker at the configuration using the ModuleAdminController and it throw an undefined index notice.

The thing is that i tested this module that is like the one i made and it didn't throw any notice.
https://github.com/PrestaShopCorp/pscartbanner

**Screenshots**

Screenshot_2020-09-26 Notice Undefined index size (500 Internal Server Error)

#### Additional information

* PrestaShop version: 1.7.6.7
* PHP version: N/A

### PR DESCRIPTION
**Pull request by bpato**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Add color picker default size value to prevent undefined error, set the key as value of the parameter name just like the others and let the original to avoid issues, fix duplicate parameter class and fix some lost double quotes.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21173.
| How to test?  | Use a ModuleAdminController with a color picker

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21174)

### CODE REVIEW

File: admin-dev/themes/default/template/helpers/options/options.tpl [hunk 1]
```diff
@@ -204,14 +204,14 @@
...
+											<textarea class="{if isset($field['autoload_rte']) && $field['autoload_rte']}rte autoload_rte{else}textarea-autosize{/if}" name="{$key}" {if isset($field['cols'])} cols="{$field['cols']}"{/if} {if isset($field['rows'])} rows="{$field['rows']}"{/if}>{$field['value']|escape:'html':'UTF-8'}</textarea>
 										</div>
 									{elseif $field['type'] == 'file'}
 										<div class="col-lg-9">{$field['file']}</div>
 									{elseif $field['type'] == 'color'}
 										<div class="col-lg-2">
 											<div class="input-group">
-												<input type="color" size="{$field['size']}" data-hex="true" {if isset($input.class)}class="{$field['class']}" {else}class="color mColorPickerInput"{/if} name="{$field['name']}" class="{if isset($field['class'])}{$field['class']}{/if}" value="{$field['value']|escape:'html':'UTF-8'}" />
+												<input type="color" size="{if isset($field['size'])}{$field['size']|intval}{else}5{/if}" data-hex="true" class="{if isset($field['class'])}{$field['class']}{else}color mColorPickerInput{/if}" name="{if isset($field['name'])}{$field['name']}{else}{$key}{/if}" value="{$field['value']|escape:'html':'UTF-8'}" />
```
Comment (NeOMakinG) [inline admin-dev/themes/default/template/helpers/options/options.tpl:214]: What is this default value of 5 doing ?

File: admin-dev/themes/default/template/helpers/options/options.tpl [hunk 2]
```diff
@@ -204,14 +204,14 @@
 										</div>
 									{elseif $field['type'] == 'textarea'}
 										<div class="col-lg-9">
-											<textarea class="{if isset($field['autoload_rte']) && $field['autoload_rte']}rte autoload_rte{else}textarea-autosize{/if}" name={$key}{if isset({$field['cols']})} cols="{$field['cols']}"{/if}{if isset({$field['rows']})} rows="{$field['rows']}"{/if}">{$field['value']|escape:'html':'UTF-8'}</textarea>
+											<textarea class="{if isset($field['autoload_rte']) && $field['autoload_rte']}rte autoload_rte{else}textarea-autosize{/if}" name="{$key}" {if isset($field['cols'])} cols="{$field['cols']}"{/if} {if isset($field['rows'])} rows="{$field['rows']}"{/if}>{$field['value']|escape:'html':'UTF-8'}</textarea>
```
Comment (Progi1984) [inline admin-dev/themes/default/template/helpers/options/options.tpl]:
```suggestion
											<textarea class="{if isset($field['autoload_rte']) && $field['autoload_rte']}rte autoload_rte{else}textarea-autosize{/if}" name="{$key}"{if isset($field['cols'])} cols="{$field['cols']}"{/if}{if isset($field['rows'])} rows="{$field['rows']}"{/if}>{$field['value']|escape:'html':'UTF-8'}</textarea>
```

File: admin-dev/themes/default/template/helpers/options/options.tpl (hunk 1 above)
Comment (bpato) [reply admin-dev/themes/default/template/helpers/options/options.tpl:214]: i don't know if its needed, i used the same default value that the others

**Review by NeOMakinG — APPROVED** [review]
lgtm

**Review by Progi1984 — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Ping @PrestaShop/prestashop-core-developers what do you think of this PR?
I did not manage to reproduce the issue with PS1.7.6.8 using this module: [pscartbanner](https://github.com/PrestaShopCorp/pscartbanner/blob/2df81d6a88d923bdb86c76083bf1e940483858e9/controllers/admin/AdminPsCartBannerController.php#L20 ) and this module: [ps_faviconnotificationbo](https://github.com/PrestaShop/ps_faviconnotificationbo/blob/dev/controllers/admin/AdminConfigureFaviconBoController.php#L20)
I have an exception using a custom module: https://github.com/bpato/test-color-picker

Thanks!

**Comment by Progi1984** [comment]
The bug fix is not illogical. I don't understand why you don't have it with ps_cartbanner.

**Review by sowbiba — APPROVED** [review]

**Comment by Progi1984** [comment]
Ping @NeOMakinG : Any idea ?

**Comment by NeOMakinG** [comment]
@Progi1984 @khouloudbelguith I think pscartbanner and ps_faviconnotificationbo has the size attribute, while the custom module from @bpato may have not this attribute, but it should not crash even if it hasn't the size value, I think this fix is relevant

**Review by PierreRambaud — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hello @bpato !

Sorry for the delay in QA. I tested : 

- Check with "test color" module that the configuration page is now OK : :heavy_check_mark: 
- Check on "pscartbanner" that the color picker still works fine : :heavy_check_mark: 

So your PR is QA :heavy_check_mark: !

Thanks !

**Comment by Progi1984** [comment]
Thanks @bpato

**Comment by idnovate** [comment]
This change should also be applied to https://github.com/bpato/PrestaShop/blob/develop/admin-dev/themes/default/template/helpers/form/form.tpl#L736-L750, right?

**Comment by NeOMakinG** [comment]
> https://github.com/bpato/PrestaShop/blob/develop/admin-dev/themes/default/template/helpers/form/form.tpl#L736-L750

Looks like yes, and it may be related to another recent issue I seen

**Pull request merged**